File: plat/js/entities/switch.js
import * as Draw from 'draw';
import Entity from '../entity.js';
import Dir from '../dir.js';
import CollisionType from '../collisiontype.js';
import World from '../world.js';
class Switch extends Entity {
    type = 'Switch';
    collidable = CollisionType.Platform;
    target;
    constructor(args) {
        super(args);
        this.target = args.properties?.Target ?? '';
    }
    draw() {
        Draw.setColor(172, 50, 50, 255);
        Draw.rect(this.pos[0], this.pos[1], this.size[0], this.size[1], false);
    }
    collide(other, dir) {
        if (dir != Dir.Up)
            return;
        World().state.entities.forEach(ent => {
            if (ent.name == this.target) {
                ent.trigger(this);
            }
        });
        this.destroyed = true;
    }
}
export { Switch };

File: csgame/entities/Switch.cs
using Slate2D;

[Spawnable]
class Switch : Entity
{
    String Target;

    public Switch(LDTKEntity ent) : base(ent)
    {
        Collidable = CollisionType.Platform;
        Layer = Layer.Background;
        Target = ent.Properties.GetValueOrDefault("Target")?.EntRef ?? "";
    }

    public override void Draw()
    {
        DC.SetColor(172, 50, 50, 255);
        DC.Rect(Pos.X, Pos.Y, Size.W, Size.H, false);

    }

    public override void Collide(Entity other, Dir dir)
    {
        if (dir != Dir.Up) return;

        foreach (var ent in Main.World.GameState.Entities)
        {
            if (ent.Id == Target) ent.Activate(this);
        }

        Destroyed = true;
    }
}